### DISCUSSION TITLE
Roborock room not working

### DISCUSSION THREAD

**Discussion by mgrefsrud** [humbertogontijo/homeassistant-roborock#203] [Q&A]
I have the Roborock integartion from HACS and the lovelace-xiaomi-vacuum-map-card set up after the guide i foun here, And i am using the send_comand platfomr choosen.
I get pin andgo to work, but when i try to starh room cleaning i get "roborock.exceptions.VacuumError: -10005: data for segment is not a number" in the HA error log. 
Any ideas?

**Comment by mgrefsrud** [comment]
Never mind i figured it out, i replaced the following.    
    service_call_schema:
      service: vacuum.send_command
      service_data:
        command: app_room_clean
        params: '[[selection]]'
        entity_id: '[[entity_id]]'
    predefined_selections:
With :
        service_call_schema:
          service: roborock.vacuum_clean_segment
          service_data:
            segments: '[[selection]]'
            entity_id: '[[entity_id]]'
        predefined_selections:

**Comment by mgrefsrud** [comment]
To be honest i do not remember what giude i used. But now the new card 
update is using the roborock integration, so i think it should work 
better.

---
Vennlig Hilsen
Morten Grefsrud

Den 2023-06-04 11:37, skrev AseKarlsson:

> Hi @mgrefsrud [1],
> I am trying the same setup (Roborock Integration and Xiaomi Vacuum Map 
> Card both from HACS).
> Where did you find the guide you mentioned in the first post?
> I cannot get the clean room etc commands be be right, it calls hte 
> xiaomi_miio/vacuum_clean_segment
> 
> --
> Reply to this email directly, view it on GitHub [2], or unsubscribe 
> [3].
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>


Links:
------
[1] https://github.com/mgrefsrud
[2] 
https://github.com/humbertogontijo/homeassistant-roborock/discussions/203#discussioncomment-6082127
[3]

**Reply by AseKarlsson** [reply]
Great!
